linkerd/app/outbound/src/sidecar.rs

```diff
@@ -74,15 +90,92 @@ impl Outbound<()> {
+                    // address, as it may consist of multiple distinct backends.
+                    // Therefore, just use the original destination address for now.
+                    // TODO(eliza): eventually, remove the logical addr field on
+                    // the policy route types when we use metadata for those
+                    // labels instead.
+                    let addr = orig_dst.0.into();
+                    match policy.protocol {
+                        policy::Protocol::Detect {
+                            ref http1,
+                            ref http2,
+                            ..
+                        } => {
+                            let routes = match version {
+                                http::Version::Http1 => http1.routes.clone(),
+                                http::Version::H2 => http2.routes.clone(),
+                            };
+                            http::Routes::Policy(http::policy::Routes::Http(
+                                http::policy::HttpRoutes {
+                                    addr,
+                                    backends: policy.backends.clone(),
+                                    routes,
+                                },
+                            ))
+                        }
+                        // TODO(eliza): what do we do here if the configured
+                        // protocol doesn't match the actual protocol for the
+                        // target? probably should make an error route instead?
+                        policy::Protocol::Http1(ref http1) => http::Routes::Policy(
+                            http::policy::Routes::Http(http::policy::HttpRoutes {
+                                addr,
+                                backends: policy.backends.clone(),
+                                routes: http1.routes.clone(),
+                            }),
+                        ),
+                        policy::Protocol::Http2(ref http2) => http::Routes::Policy(
+                            http::policy::Routes::Http(http::policy::HttpRoutes {
+                                addr,
+                                backends: policy.backends.clone(),
+                                routes: http2.routes.clone(),
+                            }),
+                        ),
+                        policy::Protocol::Grpc(ref grpc) => http::Routes::Policy(
+                            http::policy::Routes::Grpc(http::policy::GrpcRoutes {
+                                addr,
+                                backends: policy.backends.clone(),
+                                routes: grpc.routes.clone(),
+                            }),
+                        ),
+                        // TODO(eliza): if the policy's protocol is opaque, but we
+                        // are handling traffic as HTTP...that's obviously wrong.
+                        // should we panic or just fail traffic here?
```

Comment:
Let's get clearer about this condition. When would we handle traffic as HTTP but have policy protocol as opaque? Does this indicate a race of some sort? Or is this actually unreachable? It would be helpful to be clearer about this.

Reply:
I *believe* that if the policy's protocol was opaque, we won't build an HTTP stack, and won't be in this code at all. We should maybe have tests to ensure that this is the case, but changing the policy protocol should tear the whole stack down and rebuild it...